### DISCUSSION TITLE
EXR with DWAA compression not working

### DISCUSSION THREAD

**Discussion by lgander** [ImageMagick/ImageMagick#5574] [Help] [answered]
I'm using the IM plugin in Seer. 
It works fine with EXR files with ZIP compression but doesn't work with EXR files with DWAA compression.
Is there anything I could try to make it work? Any ideas?
Thanks!

**Comment by fmw42** [comment]
If it is not in the list from

`convert -list compress`

then that compression is not available.  Even if it is, it may not be available by Imagemagick for EXR.  Imagemagick has not fully coded the full EXR format as far as I know. For example it is limited to half-format EXR or was at one time.

Only docs for EXR that I know are at https://imagemagick.org/script/formats.php and https://imagemagick.org/script/high-dynamic-range.php

**Reply by lgander** [reply]
Thats bad news, as DWAA is becoming the most popular compression for EXR.

Thanks for taking the time to reply!

**Reply by fmw42** [reply]
You can make a request for an enhancement.

**Comment by snibgo** [comment]
> It works fine with EXR files with ZIP compression but doesn't work with EXR files with DWAA compression.

"Doesn't work" is not a helpful description.

Perhaps you can say how you know it doesn't work, and what error messages you get from what software, and so on.

Also give your IM version, the IM command, and any input files.

**Reply by lgander** [reply]
Sorry if I wasn't clear enough, but as I said in my first message I'm using the IM plugin in Seer. 

Seer is an application for pc that emulates the quicklook command of mac os (hitting space bar over a file displays a preview of it).
So hitting the space bar over a DOC, PDF, JPG, etc shows a preview of the DOC, PDF, JPG, etc  file.
After installing the IM plugin in Seer, the space bar over a EXR with zip compression shows it content, with DWAA it doesn't.  

Not sure if this was better than  "Works" and "Doesn't work" . Please let me know if there is any extra information you consider that might help with this.

**Reply by fmw42** [reply]
Have you consulted with the Seer people about it? If so, what did they say? Did they point to Imagemagick?

**Reply by lgander** [reply]
I've sent an email to Seer developer, he asked me to send him a couple of EXR so he can check this, I've sent one ZIP and one DWAA, but hadn't received any news from him after that.

**Comment by snibgo** [comment]
@lgander: What version of IM?

With IM v7.1.0-49, I can use IM to write an EXR with "`-compress DWAA`", and read it back. Exiftool says the EXR file has:
```
Compression                     : Unknown (8)
```
If I omit "`-compress DWAA`" then exiftool says there is no compression.

So I suppose I am successfully writing and reading DWAA. But I don't know how to verify that.

**Reply by fmw42** [reply]
exiftool says compression 0 for such file

**Reply by lgander** [reply]
I'm not sure how to check the IM version, in Seer the is a link to download the plugin folder and that's all I know about IM.
Just in case, you can download the plugin folder from this link: http://1218.io/seer/uwp-plugin.html
Or this is the direct link: http://1218.io/seer/ImageMagick.zip

If you send me the EXR i can check the compression (using Foundry Nuke) and then let you know if if its DWAA.

**Comment by snibgo** [comment]
I have uploaded two files, compressed respectively DWAA and DWAB:

http://snibgo.com/im/xdwaa.exr
http://snibgo.com/im/xdwab.exr

**Reply by lgander** [reply]
Thanks a lot for taking the time to make this tests.
Both files have the correct compression type (DWAA and DWAB).

So I guess the problem might be in Seer.

Or maybe the IM version of the Seer plugin is not updated. 
If that would be the case do you think I could try replacing some of the files in the plugin folder? (I can always roll back to the original)
I'm attaching a screengrab of the files of the plugin just in case you recognize any of this files.

Thanks again

**Reply by lgander** [reply]
Sorry, here is the screenshot:
![IM plugin files](https://user-images.githubusercontent.com/39671907/193474330-70fad217-318d-4fa8-8165-e4ae731cdc9f.png)

**Comment by snibgo** [comment, marked as answer]
I downloaded the plugin from your link. It contains convert.exe, v7.0.9-20 (which is old), which contains OpenEXR v2.3.0 (which is also old; current IM has OpenEXR v3.1.3). However, that can still write and read DWAA and DWBB.

```
f:\web\im\seer>convert ..\toes.png -colorspace RGB -compress dwaa xa.exr

f:\web\im\seer>convert ..\toes.png -colorspace RGB -compress dwab xb.exr
```
I have uploaded xa.exr and xb.exr to:

http://snibgo.com/im/xa.exr
http://snibgo.com/im/xb.exr

Please check those files.

If they are not okay, then we might blame the old versions of IM and OpenEXR, and I suggest you ask Seer to update the plugin. Another option is that you can download a binary of IM for your computer, ensuring you include "legacy components", which should give you convert.exe.

However, if the files are both valid, then I don't know what the problem is.

**Reply by lgander** [reply]
Both files have the correct compression, so as you said, this doesn't seem to be the problem.
I'll get in touch with Seer's developer and will give him this information. 

I'll get back to you when I got more information.

Thank you very much for your time and for the help.

**Reply by ccseer** [reply]
Hi,
does it mean we have to use different commands to parse files with different compression?

**Reply by fmw42** [reply]
No. I do not believe so.  I think it is the version of OpenEXR.  I cannot get IM to identify the compression with my OpenEXR 2.3.  But snibgo can with his OpenEXR 3.x

**Comment by snibgo** [comment]
Another test that may be useful: please let us see one of your EXR files that has DWAA or DWAB compression. Then we can test whether current IM (or the old IM from the plugin) can successfully read it.

You can zip the flie(s) and upload to this forum.

**Comment by fmw42** [comment]
Nevertheless, Imagemagick needs to be able to return the proper compression in identify, etc and show it in the list of compressions.  This is a bug or oversight.

**Comment by snibgo** [comment]
It does, for me (pre-built binary on Windows 8.1):
```
f:\web\im>%IMG7%magick -list compress |grep DWA
DWAA
DWAB

f:\web\im>%IMG7%magick xdwaa.exr -verbose info: |grep -i compr
  Compression: DWAA
```

**Reply by fmw42** [reply]
I just checked and my version of OpenEXR is 2.3.  So that likely explains it.

**Comment by lgander** [comment]
I've been talking with the developer yesterday and it's working now.
Thank you all for the help!

**Comment by ccseer** [comment]
Thanks for your help, guys. Appreciate it.

**Reply by fmw42** [reply]
What did they do to make it work?  Did they update OpenEXR?

**Reply by ccseer** [reply]
Yes. We updated the ImageMagick, http://1218.io/seer/ImageMagick.zip
Then everything works fine.

**Reply by ccseer** [reply]
![image](https://user-images.githubusercontent.com/15963166/193880395-64503227-64e8-4ada-a78c-a81141955f15.png)

**Reply by ccseer** [reply]
The commandline we use is still the same.
![image](https://user-images.githubusercontent.com/15963166/193880545-65cea84d-1492-4d69-8d3e-4e1e07e50e5c.png)

**Reply by ccseer** [reply]
Just made a donatioin to support your work. 
Thanks for your time.
<3
